java/flight/flight-core/src/main/java/org/apache/arrow/flight/FlightClient.java

```diff
@@ -579,6 +579,96 @@ public FlightEndpoint renewFlightEndpoint(RenewFlightEndpointRequest request, Ca
     return result;
   }
 
+  /**
+   * Request the server to extend the lifetime of a query result set.
+   *
+   * @param request The session options to set on the server.
+   * @param options Call options.
+   * @return The result containing per-value error statuses, if any.
+   */
+  public SetSessionOptionsResult setSessionOptions(SetSessionOptionsRequest request, CallOption... options) {
+    Action action = new Action(FlightConstants.SET_SESSION_OPTIONS.getType(), request.serialize().array());
+    Iterator<Result> results = doAction(action, options);
+    if (!results.hasNext()) {
+      throw CallStatus.INTERNAL
+          .withDescription("Server did not return a response")
+          .toRuntimeException();
+    }
+
+    SetSessionOptionsResult result;
+    try {
+      result = SetSessionOptionsResult.deserialize(ByteBuffer.wrap(results.next().getBody()));
+    } catch (IOException e) {
+      throw CallStatus.INTERNAL
+          .withDescription("Failed to parse server response: " + e)
+          .withCause(e)
+          .toRuntimeException();
+    }
+    results.forEachRemaining((ignored) -> {
+    });
+    return result;
+  }
+
+  /**
+   * Request the server to extend the lifetime of a query result set.
+   *
+   * @param request The (empty) GetSessionOptionsRequest.
+   * @param options Call options.
+   * @return The result containing the set of session options configured on the server.
+   */
+  public GetSessionOptionsResult getSessionOptions(GetSessionOptionsRequest request, CallOption... options) {
+    Action action = new Action(FlightConstants.GET_SESSION_OPTIONS.getType(), request.serialize().array());
+    Iterator<Result> results = doAction(action, options);
+    if (!results.hasNext()) {
+      throw CallStatus.INTERNAL
+          .withDescription("Server did not return a response")
+          .toRuntimeException();
+    }
+
+    GetSessionOptionsResult result;
+    try {
+      result = GetSessionOptionsResult.deserialize(ByteBuffer.wrap(results.next().getBody()));
+    } catch (IOException e) {
+      throw CallStatus.INTERNAL
+          .withDescription("Failed to parse server response: " + e)
+          .withCause(e)
+          .toRuntimeException();
+    }
+    results.forEachRemaining((ignored) -> {
+    });
+    return result;
+  }
+
+  /**
+   * Request the server to extend the lifetime of a query result set.
```

Comment:
The description doesn't match the method.

Reply:
Thanks.  Fixed!